backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/business_rule/event/BusinessRuleEventProcessorCommand.java

```diff
@@ -177,22 +185,44 @@ private void processEvent0(final BusinessRuleEvent event)
 			try (final IAutoCloseable ignored = Env.switchContext(tempCtx))
 			{
 				final AdMessageKey messageKey = getAdMessageKey(rule);
+
+				final StringBuilder documentSummaryBuilder = new StringBuilder();
+
+				if(!Check.isEmpty(targetRecordRef.getTableName()))
```

Comment:
i think there is no case when tablename would be empty....
but! why are we actually appending the table's name?
think about C_Invoice.... we have a lot of types like vendor invoice, vendor credit memos etc ... all those have individual windows. user has no idea that behind all that windows is actually o single table.
SO i am questioning what's the reason of appending the table name? is there in the requirements? wouldn't confuse the user more?

Reply:
> SO i am questioning what's the reason of appending the table name? is there in the requirements? 

Yes, it's in the first requirement: ``and include the table + value / Doc. No. / part of name of the entry in question``.
I could change all messages to display the table name in the message itself and establish it as a rule to have the table name/ doctype in all business rule messages. WDYT, would that be better?

Reply:
fine. if it's in the requirements, then let it be!

as a follow-up/later times, i would prefer to display the proper window name instead of table name, because that's what the user understands.